nodejs/src/stream_manager.rs

```diff
@@ -0,0 +1,632 @@
+            .to_vec(&mut cx)?
+            .into_iter()
+            .map(|v| {
+                v.downcast_or_throw::<JsString, _>(&mut cx)
+                    .map(|v| v.value(&mut cx))
+            })
+            .collect::<Result<Vec<_>, _>>()?;
+
+        let stream_result = stream_manager.create_stream_with_policy(
+            &scope_name,
+            &stream_name,
+            // TODO: Handle<JsBox<T>> auto dereferencing won't work for this (no idea why),
+            //       so manually getting the internal scaling and retention is required.
+            // https://docs.rs/neon/latest/neon/types/struct.JsBox.html#deref-behavior
+            scaling_policy.scaling.clone(),
+            retention_policy.retention.clone(),
+            match tags.len() {
+                l if l == 0 => None,
+                _ => Some(tags),
+            },
+        );
+        match stream_result {
+            Ok(t) => Ok(cx.boolean(t)),
+            Err(e) => cx.throw_error(e.to_string()),
+        }
+    }
+
+    pub fn js_update_stream_with_policy(mut cx: FunctionContext) -> JsResult<JsBoolean> {
+        let stream_manager = cx.this().downcast_or_throw::<JsBox<StreamManager>, _>(&mut cx)?;
+        let scope_name = cx.argument::<JsString>(0)?.value(&mut cx);
+        let stream_name = cx.argument::<JsString>(1)?.value(&mut cx);
+        let retention_policy = cx.argument::<JsBox<StreamRetentionPolicy>>(2)?;
+        let scaling_policy = cx.argument::<JsBox<StreamScalingPolicy>>(3)?;
+        let tags = cx
+            .argument::<JsArray>(4)?
+            .to_vec(&mut cx)?
+            .into_iter()
+            .map(|v| {
+                v.downcast_or_throw::<JsString, _>(&mut cx)
+                    .map(|v| v.value(&mut cx))
+            })
+            .collect::<Result<Vec<_>, _>>()?;
+
+        let stream_result = stream_manager.update_stream_with_policy(
+            &scope_name,
+            &stream_name,
+            scaling_policy.scaling.clone(),
+            retention_policy.retention.clone(),
+            match tags.len() {
+                l if l == 0 => None,
+                _ => Some(tags),
+            },
+        );
+        match stream_result {
+            Ok(t) => Ok(cx.boolean(t)),
+            Err(e) => cx.throw_error(e.to_string()),
+        }
+    }
+
+    pub fn js_get_stream_tags(mut cx: FunctionContext) -> JsResult<JsArray> {
+        use std::convert::TryInto;
+
+        let stream_manager = cx.this().downcast_or_throw::<JsBox<StreamManager>, _>(&mut cx)?;
+        let scope_name = cx.argument::<JsString>(0)?.value(&mut cx);
+        let stream_name = cx.argument::<JsString>(1)?.value(&mut cx);
+
+        let tags = match stream_manager.get_stream_tags(&scope_name, &stream_name) {
+            Ok(val) => match val {
+                Some(val) => val,
+                None => return Ok(cx.empty_array()),
+            },
+            Err(_err) => {
+                // TODO: display the error details.
```

Comment:
Please implement. (The display trait should provide a coherent message)

Reply:
display err with `format!`